examples/kafka/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"os"
+	"time"
+
+	kafka "github.com/segmentio/kafka-go"
+)
+
+func main() {
+	// Get the connection info from the ENV vars
+	host := os.Getenv("HOST")
+	port := os.Getenv("PORT")
+	topic := os.Getenv("TOPIC")
+
+	// And create a new kafka reader
+	reader := kafka.NewReader(kafka.ReaderConfig{
+		Brokers:  []string{host + ":" + port},
+		Topic:    topic,
+		MinBytes: 10e3,
+		MaxBytes: 10e6,
+	})
+	defer reader.Close()
+
+	// Open the /pfs/out pipe with write only permissons (the pachyderm spout will be reading at the other end of this)
+	// Note: it won't work if you try to open this with read, or read/write permissions
+	out, err := os.OpenFile("/pfs/out", os.O_WRONLY, 0644)
+	if err != nil {
+		panic(err)
+	}
+	defer out.Close()
+
+	// this is the file loop
+	for {
+		if err := func() error {
+			tw := tar.NewWriter(out)
+			defer tw.Close()
+
+			var b bytes.Buffer
+
+			// we'll use a ticker so that our spout will write files containing the kafka messages it consumed in 20ms intervals
+			tick := time.NewTicker(20 * time.Millisecond)
+			defer tick.Stop()
+			// if we wanted one file per message, we could do that by removing the for loop and the select code,
+			// so we would just read the message, and write the header and the message, and return
+
+			// this is the message loop
+			for {
+				select {
+				default:
+					// read a message and write it to the buffer
+					m, err := reader.ReadMessage(context.Background())
+					if err != nil {
+						panic(err)
```

Comment:
Minor nit, it's a little bit weird that this error gets panicked when the other ones in the function get returned (and then panicked). Would be a bit more unified to just return this one like the others and only panic in one place. It actually does have a slight semantic difference, because when you `panic` deferred functions don't get called.